### PR TITLE
Remove db.name attribute from Redis instrumentation

### LINKED ISSUE #1395: [opentelemetry-instrumentation-redis] Stop sending db.name argument or rename it to redis[0-15]
Hi! I've been recently playing with OpenTelemetry for Python (Flask) application and noticed that for Redis **db.name** argument is send to OpenTelemetry collector which seems to be a number of database (integer). This seems to be incorrect as in Redis there is no db name concept (databases are numbered from 0 to 15). Technically, it shouldn't be any problem with that but it may break some OpenTelemetry backends which expects a real DB name not a number. I have done some additional debugging and found that for node.js and .NET **db.name** argument is not send to collector. Shouldn't we have some consistency here?

**Describe your environment** 

$ python --version
Python 3.8.13
$ 
$ pip list | grep 'opentelemetry\|redis'
opentelemetry-api                        1.13.0
opentelemetry-distro                     0.34b0
opentelemetry-exporter-otlp              1.13.0
opentelemetry-exporter-otlp-proto-grpc   1.13.0
opentelemetry-exporter-otlp-proto-http   1.13.0
opentelemetry-instrumentation            0.34b0
opentelemetry-instrumentation-aws-lambda 0.34b0
opentelemetry-instrumentation-dbapi      0.34b0
opentelemetry-instrumentation-flask      0.34b0
opentelemetry-instrumentation-grpc       0.34b0
opentelemetry-instrumentation-jinja2     0.34b0
opentelemetry-instrumentation-logging    0.34b0
opentelemetry-instrumentation-redis      0.34b0
opentelemetry-instrumentation-requests   0.34b0
opentelemetry-instrumentation-sqlite3    0.34b0
opentelemetry-instrumentation-urllib     0.34b0
opentelemetry-instrumentation-urllib3    0.34b0
opentelemetry-instrumentation-wsgi       0.34b0
opentelemetry-propagator-aws-xray        1.0.1
opentelemetry-proto                      1.13.0
opentelemetry-sdk                        1.13.0
opentelemetry-semantic-conventions       0.34b0
opentelemetry-util-http                  0.34b0
redis                                    4.3.4

**Steps to reproduce**
Any Python app with connection to Redis will show this behavior.

**What is the expected behavior?**
Stop sending db.name argument or rename it to redis[0-15]

**What is the actual behavior?**
The db.name argument is send as a number of Redis database.

**Additional context**

Please see below some logs from OpenTelemetry collector for python and node.js to see a difference.

===> PYTHON EXAMPLE

ScopeSpans #0
ScopeSpans SchemaURL: 
InstrumentationScope opentelemetry.instrumentation.redis 0.34b0
Span #0
    Trace ID       : 4bc10b43ab0a0d3042f38ebbb32baef1
    Parent ID      : 79e2aed933827894
    ID             : 22f4fba607e73a33
    Name           : HMSET
    Kind           : SPAN_KIND_CLIENT
    Start time     : 2022-10-21 09:40:50.606962566 +0000 UTC
    End time       : 2022-10-21 09:40:50.609568624 +0000 UTC
    Status code    : STATUS_CODE_UNSET
    Status message : 
Attributes:
     -> db.statement: STRING(HMSET person1-hash name jane age 20)
     -> db.system: STRING(redis)
     -> db.name: INT(0)
     -> db.redis.database_index: INT(0)
     -> net.peer.name: STRING(redis-svc)
     -> net.peer.port: STRING(6379)
     -> net.transport: STRING(ip_tcp)
     -> db.redis.args_length: INT(6)

===> NODEJS EXAMPLE

ScopeSpans #0
ScopeSpans SchemaURL: 
InstrumentationScope @opentelemetry/instrumentation-redis-4 0.33.0
Span #0
    Trace ID       : 21a071f4d1d7c860ecb758398d304f60
    Parent ID      : 1bbf5328c079ceda
    ID             : 13dc47b2521f7f82
    Name           : redis-GET
    Kind           : SPAN_KIND_CLIENT
    Start time     : 2022-10-21 09:47:16.9553723 +0000 UTC
    End time       : 2022-10-21 09:47:16.957585 +0000 UTC
    Status code    : STATUS_CODE_UNSET
    Status message : 
Attributes:
     -> db.system: STRING(redis)
     -> net.peer.name: STRING(redis-svc)
     -> net.peer.port: INT(6379)
     -> db.statement: STRING(GET)
ResourceSpans #4
Resource SchemaURL: 
Resource labels:
     -> service.name: STRING(nodejs-redis)
     -> telemetry.sdk.language: STRING(nodejs)
     -> telemetry.sdk.name: STRING(opentelemetry)
     -> telemetry.sdk.version: STRING(0.24.0)
     
I am happy to contribute to it by reviewing the code fix and testing the behavior.

@svrnm @sanketmehta28

### PR DESCRIPTION
**Pull request by luke6Lh43**
# Description

There is no DB name concept in Redis (databases are numbered from 0 to 15), hence sending **db.name** attribute to OTel collector doesn't make sense. In the current implementation **db.name** attribute is defaulted to DB number. Technically, it shouldn't be any problem with that but it may break some OpenTelemetry backends which expects a real DB name not a number. There is another attribute **db.redis.database_index** which should be used for this purpose. 

After this fix, we will be also alligned with below document where is stated that for Redis **db.name** attribute shouldn't be set:

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md

Fixes #1395 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Firstly, I run following command to test the setup after the code change:

`tox -e test-instrumentation-redis`

Then, I tested that locally with some app and after removing **db.name** attribute from instrumentation, it is not exported to OTel collector as expected.

**Before the fix:**

```
InstrumentationScope opentelemetry.instrumentation.redis 0.34b0
Span #0
    Trace ID       : cf35dd1cc002f28471e83522955fa616
    Parent ID      : 0665eddd63d1811b
    ID             : 340b84683b6076c3
    Name           : RPUSH
    Kind           : SPAN_KIND_CLIENT
    Start time     : 2022-11-02 11:15:54.311276 +0000 UTC
    End time       : 2022-11-02 11:15:54.318314 +0000 UTC
    Status code    : STATUS_CODE_UNSET
    Status message : 
Attributes:
     -> db.name: Int(0)
     -> db.redis.args_length: Int(3)
     -> db.redis.database_index: Int(0)
     -> db.statement: Str(RPUSH key {"time": "2022-11-02 12:15:54", "voter_id": "4e03a47155c7252", "choice": "alone"})
     -> db.system: Str(redis)
     -> net.peer.name: Str(localhost)
     -> net.peer.port: Str(6379)
     -> net.transport: Str(ip_tcp)
Span #1
    Trace ID       : dfab74806a3f5a6eb27dd96ea5825486
    Parent ID      : 2ba23ddbdcbbb60d
    ID             : 68818dbba797fd91
    Name           : LRANGE
    Kind           : SPAN_KIND_CLIENT
    Start time     : 2022-11-02 11:15:54.335377 +0000 UTC
    End time       : 2022-11-02 11:15:54.34132 +0000 UTC
    Status code    : STATUS_CODE_UNSET
    Status message : 
Attributes:
     -> db.name: Int(0)
     -> db.redis.args_length: Int(4)
     -> db.redis.database_index: Int(0)
     -> db.statement: Str(LRANGE key 0 -1)
     -> db.system: Str(redis)
     -> net.peer.name: Str(localhost)
     -> net.peer.port: Str(6379)
     -> net.transport: Str(ip_tcp)
```

**After the fix:**

```
InstrumentationScope opentelemetry.instrumentation.redis 0.34b0
Span #0
    Trace ID       : ab2622f411379b075ec6ebf699e08525
    Parent ID      : e7ae303f79d7a23a
    ID             : bec8e7f732dd3c4b
    Name           : RPUSH
    Kind           : SPAN_KIND_CLIENT
    Start time     : 2022-11-01 10:56:13.013253 +0000 UTC
    End time       : 2022-11-01 10:56:13.019925 +0000 UTC
    Status code    : STATUS_CODE_UNSET
    Status message : 
Attributes:
     -> db.redis.args_length: Int(3)
     -> db.redis.database_index: Int(0)
     -> db.statement: Str(RPUSH key {"time": "2022-11-01 11:56:13", "voter_id": "577278715f01a5e", "choice": "alone"})
     -> db.system: Str(redis)
     -> net.peer.name: Str(localhost)
     -> net.peer.port: Str(6379)
     -> net.transport: Str(ip_tcp)
Span #1
    Trace ID       : 71bbcbb379bb5baa7a2702287bcfe401
    Parent ID      : b117d6756b142459
    ID             : 0b6621439c2bfea1
    Name           : LRANGE
    Kind           : SPAN_KIND_CLIENT
    Start time     : 2022-11-01 10:56:13.035479 +0000 UTC
    End time       : 2022-11-01 10:56:13.040153 +0000 UTC
    Status code    : STATUS_CODE_UNSET
    Status message : 
Attributes:
     -> db.redis.args_length: Int(4)
     -> db.redis.database_index: Int(0)
     -> db.statement: Str(LRANGE key 0 -1)
     -> db.system: Str(redis)
     -> net.peer.name: Str(localhost)
     -> net.peer.port: Str(6379)
     -> net.transport: Str(ip_tcp)
```

# Does This PR Require a Core Repo Change?

- No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Review by sanketmehta28 — COMMENTED** [review]
Please add the test case for this change

**Comment by luke6Lh43** [comment]
I've tried to fix docker-tests which is failing but it looks there is more generic issue... I've run it locally on my machine and getting an error related to psycopg2 library:

```
<snipped>

× Encountered error while trying to install package.
╰─> psycopg2

<snipped>
```

To be 100% sure that issue is not related to the fix I've made, I've cloned main repo and run docker-tests and got same error message... To my mind it looks like some issue with packages dependencies. Has anyone experienced similar issue recently?

**Comment by srikanthccv** [comment]
That's a regular python dep installation issue. It is probably not related to this project, but in general, you may have an issue installing the above. Are you sure your system is set up to run the psycopg2?

**Comment by luke6Lh43** [comment]
Hi. I've added another commit where I've simply removed verification of db.name attribute from test functions. @sanketmehta28 since we're removing db.name attribute from instrumentation, do we really need another test function for that or removing verification of that attribute from existing functions would be enough?

**Comment by sanketmehta28** [comment]
> 
removing the verification of the attribute and adding verification for new attribute should be sufficient

File: tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py [hunk 1]
```diff
@@ -37,7 +37,6 @@ def tearDown(self):
     def _check_span(self, span, name):
         self.assertEqual(span.name, name)
         self.assertIs(span.status.status_code, trace.StatusCode.UNSET)
-        self.assertEqual(span.attributes.get(SpanAttributes.DB_NAME), 0)
```
Comment (sanketmehta28) [inline tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py:40]: add check for attribute RD_REDIS_DATABASE_INDEX

File: tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py [hunk 2]
```diff
@@ -209,7 +208,6 @@ def tearDown(self):
...
```
Comment (sanketmehta28) [inline tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py:212]: same as above

**Review by sanketmehta28 — CHANGES_REQUESTED** [review]

File: tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py (hunk 2 above)
Comment (luke6Lh43) [reply tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py:212]: Yes, that makes sense. Will add it and commit.

File: tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py (hunk 1 above)
Comment (luke6Lh43) [reply tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py:40]: Ok, done.

File: tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py (hunk 2 above)
Comment (luke6Lh43) [reply tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py:212]: Ok, done.

**Review by sanketmehta28 — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -74,6 +74,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#1333](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1333))
 - Use resp.text instead of resp.body for Falcon 3 to avoid a deprecation warning.
   ([#1412](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1412))
+- Remove db.name attribute from Redis instrumentation
+  ([#1427](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1427))
```
Comment (srikanthccv) [inline CHANGELOG.md:78]: Move this to Unreleased section

Comment (luke6Lh43) [reply CHANGELOG.md:78]: sure, I'll do and commit again

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**